staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -328,12 +328,15 @@ func (s *store) GuaranteedUpdate(
 			// etcd in order to be sure the data in the store is equivalent to
 			// our desired serialization
 			if mustCheckData {
-				origState, err = getCurrentState()
+				origState, err := getCurrentState()
 				if err != nil {
 					return err
 				}
-				mustCheckData = false
-				continue
+				if !bytes.Equal(data, origState.data) {
```

Comment:
> How does this fix #53379?

Without this change, we will restart loop anyway.
For crds, we will fail in the second loop.

Reply:
I'm ok with the change, I just don't see how it fixes #53379... doesn't that have more to do with CRD not supporting strategic merge patch?

Reply:
> doesn't that have more to do with CRD not supporting strategic merge patch?

Yes, we have more to do.
After this change,, #53379  will be difficult to reproduce. But we do have some problem with CRD not supporting strategic merge patch.

I have changed the description:
`fix` ---->`patial fix`

Reply:
Hi, @liggitt @sttts 
I did some research.
[This](https://github.com/kubernetes/kubernetes/pull/48394/files#diff-0f11327dd3daafdfe8e326e895616f66R302) logic is designed to make make sure a no-op patch will cause object stored in etcd to upgrade to the new preferred version.
For example:
1. in etcd, we have apiversion v1alpha1
2. we `do patch -p ‘{}' `
3. in etcd, the object is stored as v1beta1

So, for CRD, we will never run into [this](https://github.com/kubernetes/kubernetes/blob/687c8d329790472ac0fa9a6f1d785fdc27710d79/staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go#L335) if scope. So, there is no second loop for CRDs, #53379 is totally fixed(or shadowed) as far as I can see.